src/dbnode/encoding/m3tsz/encoder.go

```diff
@@ -467,64 +260,35 @@ func (enc *encoder) writeIntSigMult(sig, mult uint8, floatChanged bool) {
 	}
 }
 
-// trackNewSig gets the new number of significant bits given the
-// number of significant bits of the current diff. It takes into
-// account thresholds to try and find a value that's best for the
-// current data
-func (enc *encoder) trackNewSig(numSig uint8) uint8 {
-	newSig := enc.numSig
-
-	if numSig > enc.numSig {
-		newSig = numSig
-	} else if enc.numSig-numSig >= sigDiffThreshold {
-		if enc.numLowerSig == 0 {
-			enc.curHighestLowerSig = numSig
-		} else if numSig > enc.curHighestLowerSig {
-			enc.curHighestLowerSig = numSig
-		}
-
-		enc.numLowerSig++
-		if enc.numLowerSig >= sigRepeatThreshold {
-			newSig = enc.curHighestLowerSig
-			enc.numLowerSig = 0
-		}
-
-	} else {
-		enc.numLowerSig = 0
-	}
-
-	return newSig
-}
-
 func (enc *encoder) newBuffer(capacity int) checked.Bytes {
 	if bytesPool := enc.opts.BytesPool(); bytesPool != nil {
 		return bytesPool.Get(capacity)
 	}
 	return checked.NewBytes(make([]byte, 0, capacity), nil)
 }
 
+// Reset resets the encoder for reuse.
 func (enc *encoder) Reset(start time.Time, capacity int) {
 	enc.reset(start, enc.newBuffer(capacity))
 }
 
 func (enc *encoder) reset(start time.Time, bytes checked.Bytes) {
 	enc.os.Reset(bytes)
-	enc.t = start
-	enc.dt = 0
-	enc.vb = 0
-	enc.xor = 0
+
+	timeUnit := initialTimeUnit(start, enc.opts.DefaultTimeUnit())
+	enc.tsEncoderState = NewTimestampEncoder(start, timeUnit, enc.opts)
+
+	enc.xorEncoderState = XOREncoder{}
```

Comment:
Why create a new encoder instead of having a reset on existing XOREncoder?

Reply:
They're all non-pointers so this is basically "free" / equivalent to a memclear operation if that makes sense.

Reply:
Guess these aren't pooled eh? Should we think of pooling them, won't a whole bunch be created almost at the same time every block period?